Tests/SonarQube.MSBuild.Tasks.IntegrationTests/TargetsTests/BuildWrapperTargetsTests.cs

```diff
@@ -0,0 +1,182 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="BuildWrapperTargetsTests.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.Build.Construction;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.IO;
+using TestUtilities;
+
+namespace SonarQube.MSBuild.Tasks.IntegrationTests.TargetsTests
+{
+    [TestClass]
+    public class BuildWrapperTargetsTests
+    {
+        public TestContext TestContext { get; set; }
+
+        #region Tests
+
+        [TestMethod]
+        public void SQBeforeClCompile_TempPathNotSet_NotExecuted()
```

Comment:
This test and next one check the preconditions for the task to be executed i.e. don't execute if the temp path isn't set, of if the user has set the "off" switch $(SQSkipBuildWrapper)